iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -619,6 +622,41 @@ PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
     return maybePublisherPortData;
 }
 
+cxx::expected<PublisherPortRouDiType::MemberType_t*, PortPoolError>
+PortManager::acquirePublisherPortDataWithoutDiscovery(const capro::ServiceDescription& service,
+                                                      const popo::PublisherOptions& publisherOptions,
+                                                      const RuntimeName_t& runtimeName,
+                                                      mepoo::MemoryManager* const payloadDataSegmentMemoryManager,
+                                                      const PortConfigInfo& portConfigInfo) noexcept
+{
+    if (doesViolateCommunicationPolicy<iox::build::CommunicationPolicy>(service).and_then(
```

Comment:
Seeing this, I think when a port is requested, RouDi should check whether the `ServiceDescription` matches one of the internal services like introspection or service discovery and deny the port creation if it is an internal service. Currently a user could create a publisher with the `ServiceDescription` of the service discovery and confuse all users of it.

Could you please create an github issues for this?

Reply:
Hehehe but then we close the door for iceoryx spoofing attacks ;)

Reply:
Yes, essentially we blacklist some service descriptions which are internally used.